File: OLSource1/v1/VS_officedev/codesnippet/VisualBasic/customizing-ui-features-by-using-extensibility-interfaces_2.vb
    Friend taskPaneHelper1 As TaskPaneHelper

    Protected Overrides Function RequestService( _
        ByVal serviceGuid As Guid) As Object

        If (serviceGuid = GetType(Office.ICustomTaskPaneConsumer).GUID) Then
            If (taskPaneHelper1 Is Nothing) Then
                taskPaneHelper1 = New TaskPaneHelper()
            End If
            Return taskPaneHelper1
        End If

        Return MyBase.RequestService(serviceGuid)
    End Function

File: OLSource1/v1/VS_officedev/codesnippet/CSharp/customizing-ui-features-by-using-extensibility-interfaces_2.cs
        internal TaskPaneHelper taskPaneHelper1;

        protected override object RequestService(Guid serviceGuid)
        {
            if (serviceGuid == typeof(Office.ICustomTaskPaneConsumer).GUID)
            {
                if (taskPaneHelper1 == null)
                {
                    taskPaneHelper1 = new TaskPaneHelper();
                }
                return taskPaneHelper1;
            }

            return base.RequestService(serviceGuid);
        }